### PR TITLE
fix: companion commenting and tracking

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Fixed the query key as the comparison is failing due to the last item of the key being an array. Fixed by spreading the array.
- Moved the tracking to instead happen right away instead of waiting in the callback since all new comments must be tracked.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1516 #done

### CODE REVIEW

File: packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx [hunk 1]
```diff
@@ -129,6 +133,12 @@ export function CommentMarkdownInput({
 
     if (!editCommentId) {
       updatePostCache(client, postId, { numComments: post.numComments + 1 });
+
+      trackEvent(
```
Comment (rebelchris) [inline packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:137]: Does this mean we missed some tracking events before from companion?

**Review by rebelchris — APPROVED** [review]
Just some secondary concerns.
@idoshamun Just to note I think we might have miss some tracking events from companion.

File: packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:137]: Only after the migration we did from Modal to CommentBox.

Comment (idoshamun) [reply packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:137]: Why don't we send this event from the component itself? The fact we have to add everywhere is a bit messy

**Review by idoshamun — COMMENTED** [review]
@sshanzel have you validated that it works locally?

**Comment by sshanzel** [comment]
> @sshanzel have you validated that it works locally?

Yes, I installed companion locally too. Then tested the normal commenting and both ends have worked well 🚀

File: packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:137]: Yes, this is the fix now, we are sending it from the component itself than the callback.

Comment (rebelchris) [reply packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:137]: @idoshamun Just mentioned you as I'm not sure data team needs to be aware of these things?

Comment (idoshamun) [reply packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:137]: Thanks. Since we don't have so many companion users it's fine :)

**Pull request merged**